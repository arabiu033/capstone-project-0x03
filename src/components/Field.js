import PropTypes from 'prop-types';
import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchLaureates } from '../redux/space/laureatesSlice';
import { fieldLaureates } from '../redux/space/nobelSlice';
import { images } from './Card';
import arrow from '../assets/images/arrow.png';

const Field = (props) => {
  const { field, setTitle } = props;
  const dispatch = useDispatch();

  // Make necessary dispatches upon user click
  const handleClick = (id, count) => {
    setTitle(() => 'Laureates');
    dispatch(fieldLaureates([id, count]));
    dispatch(fetchLaureates({ id, count: count > 100 ? 50 : count }));
  };

  return (
    <Link to="/details" onClick={() => handleClick(field.id, field.count)}>
      <article type="button" className="field">
        <img src={arrow} alt="Field" className="field-arrow" />
        <img src={images[field.id]} alt="Icon" className="field-image" />
        <div className="field-detail">
          <h3 className="field-name">{field.name}</h3>
          <h3 className="field-count">{field.count}</h3>
        </div>
      </article>
    </Link>
  );
};

// PropTypes reqquired
Field.propTypes = {
  field: PropTypes.exact({
    id: PropTypes.string,
    name: PropTypes.string,
    count: PropTypes.number,
  }).isRequired,
  setTitle: PropTypes.func.isRequired,
};
export default Field;
