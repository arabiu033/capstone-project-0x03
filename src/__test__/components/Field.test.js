import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';

import Field from '../../components/Field';

test('renders correctly Header component', () => {
  const fie = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Field field={{id: 'title', name:'', count: 0}} setTitle={() => 'g'}/>
      </Provider>
    </BrowserRouter>
  ).toJSON();
  expect(fie).toMatchSnapshot();
});