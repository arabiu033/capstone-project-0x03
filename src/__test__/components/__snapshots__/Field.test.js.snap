// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Header component 1`] = `
<a
  href="/details"
  onClick={[Function]}
>
  <article
    className="field"
    type="button"
  >
    <img
      alt="Field"
      className="field-arrow"
      src="arrow.png"
    />
    <img
      alt="Icon"
      className="field-image"
    />
    <div
      className="field-detail"
    >
      <h3
        className="field-name"
      />
      <h3
        className="field-count"
      >
        0
      </h3>
    </div>
  </article>
</a>
`;
